openslides_backend/services/datastore/adapter.py

```diff
@@ -57,10 +57,17 @@ def retrieve(self, command: commands.Command) -> DatastoreResponse:
                 payload.get("error") if isinstance(payload, dict) else None
             )
             if additional_error_message is not None:
-                if (
-                    additional_error_message.get("type_verbose")
-                    == "MODEL_DOES_NOT_EXIST"
-                ):
+                type_verbose = additional_error_message.get("type_verbose")
+                if type_verbose == "MODEL_LOCKED":
+                    raise DatastoreLockedException(
+                        " ".join(
+                            (
+                                error_message,
+                                f"Model '{additional_error_message.get('key')}' raises {type_verbose} error.",
+                            )
+                        )
+                    )
+                elif type_verbose == "MODEL_DOES_NOT_EXIST":
```

Comment:
Maybe add a ModelDoesNotExist exception?

Reply:
Why would you do that? There is only one more step in handling (this kind of) actions: catching and transfering them to the ("unified") ActionException. And this is done on the most down level as possible (ServiceException), because they are all handled the same.
A diversification is in my opinion only necessary, where we need to distinguish them like in the locked case with the DatastoreLockedException. See also my other comment.